application/config/config.php

```diff
@@ -0,0 +1,461 @@
+<?php if (!defined('BASEPATH')) exit('No direct script access allowed');
+
+// config.php from project source code - keep line for Ansible tests
+
+// TODO: Why are project_shared_path and root_dir
+// determined differently?
+//   $root_dir = '{{ current_source_symlink }}';
+
+// Load .env file into environment
+$root_dir = dirname(__DIR__, 2);
+require_once($root_dir . "/vendor/autoload.php");
+
+/**
+ * Use Dotenv to set required environment variables and load .env file in root
+ */
+if (file_exists($root_dir . '/.env')) {
+    $dotenv = new \Dotenv\Dotenv($root_dir);
+    $dotenv->load();
+}
+$project_shared_path = getenv('PROJECT_SHARED_PATH') ?: '/var/www/app';
+$config['project_shared_path'] = $project_shared_path;
+$config['download_dir'] = $project_shared_path . '/downloads';
+$config['archive_dir'] = $project_shared_path . '/archive';
+$config['docs_path'] = 'https://raw.githubusercontent.com/GSA/project-open-data-dashboard/master/documentation/';
+
+$config['s3_bucket'] = getenv('S3_BUCKET') ?: '';
+$config['s3_prefix'] = getenv('S3_PREFIX') ?: '';
+
+$config['import_active'] = true;
+$config['show_all_offices'] = false;
+$config['max_remote_size'] = 500000000;
+
+$config['google_analytics_id'] = ''; // UA-xxxxxxx-xx
+$config['google_analytics_domain'] = ''; // domain.com
+
+// Set local time zone
+date_default_timezone_set(getenv('TIMEZONE') ?: 'America/New_York');
+
+$config['tmp_csv_import'] = $project_shared_path . '/downloads/import.csv';
+$config['pre_approved_admins'] = explode(",", strtolower(getenv('PRE_APPROVED_ADMINS')));
+
+// `use_local_storage`: Use the local filesystem for uploads, or use S3
+// We default to TRUE, setting to FALSE only if env is 
+// is explicitly FALSE (case insenstive)
+$config['use_local_storage'] = TRUE;
+if (
+    !empty(getenv('USE_LOCAL_STORAGE')) and
+    is_string(getenv('USE_LOCAL_STORAGE')) and 
+    (strtolower(getenv('USE_LOCAL_STORAGE')) == 'false')
+   ) {
+    $config['use_local_storage'] = FALSE;
+    log_message('debug', 'Setting use_local_storage to FALSE');
+}
+
+/*
+|--------------------------------------------------------------------------
+| Base Site URL
+|--------------------------------------------------------------------------
+|
+| URL to your CodeIgniter root. Typically this will be your base URL,
+| WITH a trailing slash:
+|
+|	http://example.com/
+|
+| If this is not set then CodeIgniter will guess the protocol, domain and
+| path to your installation.
+|
+*/
+$protocol = getenv('CONTENT_PROTOCOL') ?: 'http';
+
+$default_host = getenv('DEFAULT_HOST') ?: 'localhost';
+if (isset($_SERVER['HTTP_HOST']) && $_SERVER['HTTP_HOST']) {
+    $default_host = $_SERVER['HTTP_HOST'];
+}
```

Comment:
Interesting point. I think pulling the value from DEFAULT_HOST env will work, since that can be set correctly on a per-environment basis. 

Some interesting links: http://shiflett.org/blog/2006/server-name-versus-http-host
https://stackoverflow.com/questions/1459739/php-serverhttp-host-vs-serverserver-name-am-i-understanding-the-ma

The latter agrees on checking an array.

At least if this is wrong it'll break immediately and obviously.